tests/unit/src/main/java/org/mule/tck/junit4/matcher/ValueMatcher.java

```diff
@@ -96,16 +123,18 @@ public ValueMatcher withPartName(String partName) {
    * @return the contribute {@link ValueMatcher}
    */
   public ValueMatcher withChilds(ValueMatcher... valueMatcher) {
-    this.valueMatchers = valueMatcher;
-    descriptionBuilder.append(format(", with child values: [%s]", stream(valueMatchers)
+    this.childValues = valueMatcher;
+    descriptionBuilder.append(format(", with child values that %s", stream(childValues)
         .map(Matcher::toString)
         .collect(joining(", "))));
     return this;
   }
 
-  private void validateEquals(String actual, String expected) {
-    if (actual != null && !expected.equals(actual)) {
-      throw new RuntimeException(String.format("Assertion Error - Actual: '%s' Expected: '%s'", actual, expected));
+  private <T> void assertThat(T expected, Matcher<T> actual) {
+    if (actual != null) {
```

Comment:
Wrong name, `expected` should be `actual` and vice versa.
And the lack of the `Matcher` means that that value should not be compared